GitUI/CommandsDialogs/FormFileHistoryController.cs

```diff
@@ -0,0 +1,54 @@
+using System.IO;
+using System.Text;
+using GitCommands.Utils;
+using JetBrains.Annotations;
+
+namespace GitUI.CommandsDialogs
+{
+    public sealed class FormFileHistoryController
+    {
+        /// <summary>
+        /// Gets the exact case used on the file system for an existing file or directory.
+        /// </summary>
+        /// <param name="path">A relative or absolute path.</param>
+        /// <param name="exactPath">The full path using the correct case if the path exists.  Otherwise, null.</param>
+        /// <returns>True if the exact path was found.  False otherwise.</returns>
+        /// <remarks>
+        /// This supports drive-lettered paths and UNC paths, but a UNC root
+        /// will be returned in lowercase (e.g., \\server\share).
+        /// </remarks>
+        [ContractAnnotation("=>false,exactPath:null")]
+        [ContractAnnotation("=>true,exactPath:notnull")]
+        [ContractAnnotation("path:null=>false,exactPath:null")]
+        public bool TryGetExactPath(string path, out string exactPath)
```

Comment:
I don't think this method belongs on a class called `FormFileHistoryController`. Can we move it to [`PathUtil`](https://github.com/gitextensions/gitextensions/blob/master/GitCommands/PathUtil.cs)?

Reply:
> I don't think this method belongs on a class called `FormFileHistoryController`.

Given that the form is the only one consumer of the method the controller is kind of logical place for it. At the same time I share your sentiment (to a degree).

> Can we move it to [`PathUtil`](https://github.com/gitextensions/gitextensions/blob/master/GitCommands/PathUtil.cs)?

I'm not sure how... This feels like a chicken and an egg problem - for GitCommands project to stay OS-agnostic it shouldn't have this method.

Suggestions?

Reply:
I suppose we can do what Xamarin always suggested - have platform-specific projects that have OS-specific dependencies and provide implementations in OS-specific way.
This way our code won't be peppered with `if (Windows) { ... } else if (Linux) { ... }`

Reply:
Merging as is, we can revisit this in the future.